core/include/vecmem/containers/impl/jagged_device_vector_iterator.ipp

```diff
@@ -7,194 +7,152 @@
-   VECMEM_HOST_AND_DEVICE
-   jagged_device_vector_iterator< TYPE >
-   jagged_device_vector_iterator< TYPE >::operator+( difference_type n ) const {
-
-      return jagged_device_vector_iterator( m_ptr + n );
-   }
-
-   template< typename TYPE >
-   VECMEM_HOST_AND_DEVICE
-   jagged_device_vector_iterator< TYPE >&
-   jagged_device_vector_iterator< TYPE >::operator+=( difference_type n ) {
-
-      m_ptr += n;
-      return *this;
-   }
-
-   template< typename TYPE >
-   VECMEM_HOST_AND_DEVICE
-   jagged_device_vector_iterator< TYPE >
-   jagged_device_vector_iterator< TYPE >::operator-( difference_type n ) const {
-
-      return jagged_device_vector_iterator( m_ptr - n );
-   }
-
-   template< typename TYPE >
-   VECMEM_HOST_AND_DEVICE
-   jagged_device_vector_iterator< TYPE >&
-   jagged_device_vector_iterator< TYPE >::operator-=( difference_type n ) {
-
-      m_ptr -= n;
-      return *this;
-   }
-
-   template< typename TYPE >
-   VECMEM_HOST_AND_DEVICE
-   bool jagged_device_vector_iterator< TYPE >::
-   operator==( const jagged_device_vector_iterator& other ) const {
-
-      return ( m_ptr == other.m_ptr );
-   }
-
-   template< typename TYPE >
-   VECMEM_HOST_AND_DEVICE
-   bool jagged_device_vector_iterator< TYPE >::
-   operator!=( const jagged_device_vector_iterator& other ) const {
-
-      return !( *this == other );
-   }
-
-} } // namespace vecmem::details
+namespace vecmem {
+namespace details {
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE jagged_device_vector_iterator<TYPE>::pointer::pointer(
+    const data_pointer data)
+    : m_vec(*data) {}
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE typename jagged_device_vector_iterator<TYPE>::value_type*
+jagged_device_vector_iterator<TYPE>::pointer::operator->() {
+    return &m_vec;
+}
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE const typename jagged_device_vector_iterator<
+    TYPE>::value_type*
+jagged_device_vector_iterator<TYPE>::pointer::operator->() const {
+    return &m_vec;
+}
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE
+jagged_device_vector_iterator<TYPE>::jagged_device_vector_iterator()
+    : m_ptr(nullptr) {}
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE
+jagged_device_vector_iterator<TYPE>::jagged_device_vector_iterator(
+    data_pointer data)
+    : m_ptr(data) {}
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE
+jagged_device_vector_iterator<TYPE>::jagged_device_vector_iterator(
+    const jagged_device_vector_iterator& parent)
+    : m_ptr(parent.m_ptr) {}
+
+template <typename TYPE>
+template <typename T>
+VECMEM_HOST_AND_DEVICE
+jagged_device_vector_iterator<TYPE>::jagged_device_vector_iterator(
+    const jagged_device_vector_iterator<T>& parent)
+    : m_ptr(parent.m_ptr) {}
+
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE jagged_device_vector_iterator<TYPE>&
+jagged_device_vector_iterator<TYPE>::operator=(
+    const jagged_device_vector_iterator& rhs) {
+    // Check if anything needs to be done.
```

Comment:
Again a very personal note. I really don't like the lack of empty lines before long code blocks inside of a function. For single-line functions it's fine to have them like this. But for long functions it makes it hard (for me at least) to see where the function declaration ends, and where the function body begins.

Is there a setting that could help with this?

Reply:
I don't think so, I couldn't find anything, unfortunately.

Reply:
Scratch that, maybe `KeepEmptyLinesAtTheStartOfBlocks` can do it.